salt/_modules/metalk8s_cordon.py

```diff
@@ -97,26 +39,32 @@ def node_uncordon(node_name, **kwargs):
 
     CLI Examples::
 
-        salt '*' kubernetes.node_uncordon  node_name="minikube"
+        salt '*' kubernetes.node_uncordon node_name="minikube"
     '''
     # k-c-m 1.12 adds taint `node.kubernetes.io/unschedulable` when cordonning
     # a node but k-c-m 1.11 does not remove this taint when uncordonning so,
     # we need to manually remove it when necessary by patching taints.
 
-    all_node_taints = __salt__['metalk8s_kubernetes.node_taints'](
-        node_name, **kwargs
-    )
+    all_node_taints = __salt__['metalk8s_kubernetes.get_object'](
+        name=node_name,
+        kind='Node',
+        apiVersion='v1',
+        **kwargs
+    )['spec'].get('taints', [])
+
     key_to_remove = "node.kubernetes.io/unschedulable"
 
     if key_to_remove in (taint['key'] for taint in all_node_taints):
         new_taints = [
             taint for taint in all_node_taints
             if taint['key'] != key_to_remove
         ]
-        __salt__['metalk8s_kubernetes.node_set_taints'](
-            node_name=node_name,
-            taints=new_taints,
+        __salt__['metalk8s_kubernetes.update_object'](
```

Comment:
We make two PATCH calls, on the same object. Since we can use the power of [Strategic Merge Patch](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md), let's remove this `get taints` > `check if unschedulable taint is there` >  `remove it` > `set spec.unschedulable` flow, and replace it with a single PATCH call:

```python
return __salt__['metalk8s_kubernetes.update_object'](
    name=node_name,
    kind='Node',
    apiVersion='v1',
    patch={
        'spec': {
            'taints': [{
                'key': 'node.kubernetes.io/unschedulable',
                '$patch': 'delete',
            }],
            'unschedulable': False,
        },
    },
    **kwargs
)
```

Reply:
Hmm since it's only a workaround that should be removed in other branch I prefer to not change this.

Reply:
The workaround is just removing a few lines in the `patch` object, not worth doing two API calls.

Reply:
Yes ok it's doable, personnally don't really care I will change it